### PR TITLE
feat: support prettier

### PR DESCRIPTION
**Pull request by thilllon**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [ ] Bugfix
-   [x] Feature
-   [ ] Code style update (formatting, local variables)
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Build related changes
-   [ ] CI related changes
-   [ ] Documentation content changes
-   [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

This PR is one part of update roadmap of this libarary. https://github.com/woowabros/nestjs-library-config/issues/121

Purpose of the PR is resolving conflicts between autogenerated content using `@nestjs-library/confg`'s scanner and existing content, by providing additional `formatting` option, if user wants it.

A new option `format` is given which formats the output file with `prettier`, and it seeks prettier config file using `cosmiconfig`. We should note that `cosmiconfig` is **_default configuration file finder_** of `prettier` internally. https://github.com/prettier/prettier/blob/2bd131fbd29e94e989e64904f1a628877d38ae3d/package.json#L52C18-L52C18

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

This PR supports backward compatibility and there is no changes unless no additional options are given.

### CODE REVIEW

File: example/config-scanner.runner.sh
Comment (thilllon) [inline example/config-scanner.runner.sh:1]: this file is to run example to validate that scanner's formatting option works properly

File: example/project_1/README.md
Comment (thilllon) [inline example/project_1/README.md:1]: file is formatted by prettier.

File: example/project_1/tsconfig.json
Comment (thilllon) [inline example/project_1/tsconfig.json:1]: it is generated via `tsc --init`
It is intended to be remained as initially generated.

File: example/project_2/.prettierrc
Comment (thilllon) [inline example/project_2/.prettierrc:1]: different from `project_1`'s prettier configuration

File: src/lib/config-scanner.ts [hunk 1]
```diff
@@ -54,23 +70,24 @@ export class ConfigScanner {
...
     private environmentMap: Map<string, EnvironmentMapValue>;
+    private format: boolean;
 
     constructor(configScannerOptions: ConfigScannerOptions) {
         this.sourceRoot = configScannerOptions.sourceRoot;
         this.outputPath = configScannerOptions.outputDirectory ?? path.parse(this.sourceRoot).dir;
-
         this.filename = configScannerOptions.filename ?? MARKDOWN_FORMAT.defaultFileName;
-        this.checkFileFormat(this.filename);
+        this.format = configScannerOptions.format ?? false;
```
Comment (thilllon) [inline src/lib/config-scanner.ts:79]: should have `false` by default to support backward-compatibility

**Comment by jiho-kr** [comment]
i left a comment on https://github.com/woowabros/nestjs-library-config/issues/121.

**Pull request closed without merging**